### PR TITLE
SERVER-93200 fix windows incremental builds

### PR DESCRIPTION
**Pull request by dmoody256**

### CODE REVIEW

File: rules_poetry/defs.bzl [hunk 1]
```diff
@@ -193,61 +159,38 @@ download_wheel = rule(
...
+        wheel_info.pkg + " ; " + str(wheel_info.marker)
+    ]
 
-    ctx.actions.run_shell(
-        command = "$1 install --force-reinstall --upgrade $2 --no-compile --target=$3 \"$5 ; $4\"",
-        # second portion of the .pydistutils.cfg workaround described above
-        env = dict(deterministic_env().items() + [("HOME", setup_cfg.dirname)]),
-        inputs = ctx.files.wheel + [setup_cfg],
-        outputs = [installed_wheel],
+    ctx.actions.run(
```
Comment (zackwintermdb) [inline rules_poetry/defs.bzl:184]: Nice! replacing the run_shell too

File: rules_poetry/defs.bzl [hunk 2]
```diff
@@ -110,29 +89,17 @@ COMMON_ARGS = [
...
-        tools = depset(direct = [ctx.executable._pip])
-    else:
-        executable = runtime.interpreter.path
-        inputs = depset([requirements], transitive = [runtime.files])
-        tools = depset(direct = [runtime.interpreter,
-                                 ctx.executable._pip], transitive = [runtime.files])
-
-    pip_path = ctx.executable._pip.path
+    python = ctx.toolchains["@bazel_tools//tools/python:toolchain_type"].py3_runtime
+    destination = ctx.actions.declare_directory("wheels/%s" % ctx.attr.name)
```
Comment (zackwintermdb) [inline rules_poetry/defs.bzl:99]: Does this declare_directory still cause issues?

**Review by zackwintermdb — APPROVED** [review]
This was a lot more of an involved change than I expected, but this looks way cleaner than before

File: rules_poetry/defs.bzl (hunk 2 above)
Comment (dmoody256) [reply rules_poetry/defs.bzl:99]: no even though its declared (it has to be to install the files to the location), the return output from the rule is just the zipfile

**Pull request merged**